tests/test_sockets.py

```diff
@@ -307,6 +318,29 @@ def serve() -> None:
         server_sock.close()
         assert client_addr[0] == expected_client_addr
 
+    @pytest.mark.skipif(
+        sys.implementation.name == "pypy",
+        reason=(
+            "gc.get_referrers is broken on PyPy see "
+            "https://github.com/pypy/pypy/issues/5075"
+        ),
+    )
+    async def test_happy_eyeballs_refcycles(self) -> None:
+        """
+        Test derived from https://github.com/python/cpython/pull/124859
+        """
+        ip = "127.0.0.1"
+        port = ephemeral_port_reserve.reserve(ip=ip)
+        exc = None
+        try:
+            async with await connect_tcp(ip, port):
+                pass
+        except OSError as e:
+            exc = e.__cause__
+
+        assert isinstance(exc, OSError)
+        assert gc.get_referrers(exc) == no_other_refs()
```

Comment:
this is failing on trio for <3.13 I think it's an issue related to trio KI handling and fixed in https://peps.python.org/pep-0667/